### ISSUE TITLE
[Bug]: Setting enable_match=True on non-varchar field types will not result in an error when creating a collection.

### ISSUE THREAD

**Issue by zhuwenxing** [milvus-io/milvus#36685]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:master-20240920-4b077e1b-amd64
- Deployment mode(standalone or cluster):both
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Test code
```
    def test_query_text_match_with_unsupported_fields(self):
        """
        target: test enable text match with unsupported field
        method: 1. enable text match in unsupported field
                2. create collection
        expected: create collection failed and return error
        """
        analyzer_params = {
            "tokenizer": "default",
        }
        dim = 128
        default_fields = [
            FieldSchema(name="id", dtype=DataType.INT64, is_primary=True),
            FieldSchema(
                name="title",
                dtype=DataType.VARCHAR,
                max_length=65535,
                enable_match=True,
                analyzer_params=analyzer_params,
            ),
            FieldSchema(
                name="overview",
                dtype=DataType.VARCHAR,
                max_length=65535,
                enable_match=True,
                analyzer_params=analyzer_params,
            ),
            FieldSchema(
                name="age",
                dtype=DataType.INT64,
                enable_match=True,
                analyzer_params=analyzer_params,
            ),

            FieldSchema(name="emb", dtype=DataType.FLOAT_VECTOR, dim=dim),
        ]
        default_schema = CollectionSchema(
            fields=default_fields, description="test collection"
        )
        collection_w = self.init_collection_wrap(
            name=cf.gen_unique_str(prefix),
            schema=default_schema,
        )
        log.info(f"collection_w {collection_w.describe()}")
        data = [
            {
                "id": i,
                "title": fake_en.word().lower(),
                "overview": fake_en.text().lower(),
                "age": i,
                "emb": cf.gen_vectors(1, dim)[0],
            }
            for i in range(1000)
        ]
        collection_w.insert(data)
        collection_w.create_index(
            "emb",
            {"index_type": "IVF_SQ8", "metric_type": "L2", "params": {"nlist": 64}},
        )
        collection_w.load()
        expr = "age < 10"
        res, _ = collection_w.query(expr=expr, output_fields=["id", "age"])
        log.info(f"res len {len(res)}, {res}")
        expr = "TextMatch(age, '10')"
        res, _ = collection_w.query(expr=expr, output_fields=["id", "age"])
        log.info(f"res len {len(res)}, {res}")
```

```
[2024-10-08 14:50:56 - INFO - ci_test]: collection_w ({'collection_name': 'query_dopvRrrb', 'auto_id': False, 'num_shards': 1, 'description': 'test collection', 'fields': [{'field_id': 100, 'name': 'id', 'description': '', 'type': <DataType.INT64: 5>, 'params': {}, 'is_primary': True}, {'field_id': 101, 'name': 'title', 'description': '', 'type': <DataType.VARCHAR: 21>, 'params': {'max_length': 65535, 'enable_match': 'true', 'analyzer_params': '{"tokenizer":"default"}'}}, {'field_id': 102, 'name': 'overview', 'description': '', 'type': <DataType.VARCHAR: 21>, 'params': {'max_length': 65535, 'enable_match': 'true', 'analyzer_params': '{"tokenizer":"default"}'}}, {'field_id': 103, 'name': 'age', 'description': '', 'type': <DataType.INT64: 5>, 'params': {}}, {'field_id': 104, 'name': 'emb', 'description': '', 'type': <DataType.FLOAT_VECTOR: 101>, 'params': {'dim': 128}}], 'aliases': [], 'collection_id': 452788998742922683, 'consistency_level': 0, 'properties': {}, 'num_partitions': 1, 'enable_dynamic_field': False}, True) (test_query.py:5598)
[2024-10-08 14:50:58 - INFO - ci_test]: res len 10, data: ["{'id': 0, 'age': 0}", "{'id': 1, 'age': 1}", "{'id': 2, 'age': 2}", "{'id': 3, 'age': 3}", "{'id': 4, 'age': 4}", "{'id': 5, 'age': 5}", "{'id': 6, 'age': 6}", "{'id': 7, 'age': 7}", "{'id': 8, 'age': 8}", "{'id': 9, 'age': 9}"]  (test_query.py:5618)
[2024-10-08 14:50:58 - ERROR - pymilvus.decorators]: RPC error: [query], <MilvusException: (code=1100, message=failed to create query plan: cannot parse expression: TextMatch(age, '10'), error: text match operation on non-string is unsupported: invalid parameter)>, <Time:{'RPC start': '2024-10-08 14:50:58.083859', 'RPC error': '2024-10-08 14:50:58.101297'}> (decorators.py:140)
[2024-10-08 14:50:58 - ERROR - ci_test]: Traceback (most recent call last):
  File "/Users/zilliz/workspace/milvus/tests/python_client/utils/api_request.py", line 32, in inner_wrapper
    res = func(*args, **_kwargs)
  File "/Users/zilliz/workspace/milvus/tests/python_client/utils/api_request.py", line 63, in api_request
    return func(*arg, **kwargs)
  File "/Users/zilliz/opt/anaconda3/envs/text_match/lib/python3.10/site-packages/pymilvus/orm/collection.py", line 1076, in query
    return conn.query(
  File "/Users/zilliz/opt/anaconda3/envs/text_match/lib/python3.10/site-packages/pymilvus/decorators.py", line 141, in handler
    raise e from e
  File "/Users/zilliz/opt/anaconda3/envs/text_match/lib/python3.10/site-packages/pymilvus/decorators.py", line 137, in handler
    return func(*args, **kwargs)
  File "/Users/zilliz/opt/anaconda3/envs/text_match/lib/python3.10/site-packages/pymilvus/decorators.py", line 176, in handler
    return func(self, *args, **kwargs)
  File "/Users/zilliz/opt/anaconda3/envs/text_match/lib/python3.10/site-packages/pymilvus/decorators.py", line 116, in handler
    raise e from e
  File "/Users/zilliz/opt/anaconda3/envs/text_match/lib/python3.10/site-packages/pymilvus/decorators.py", line 86, in handler
    return func(*args, **kwargs)
  File "/Users/zilliz/opt/anaconda3/envs/text_match/lib/python3.10/site-packages/pymilvus/client/grpc_handler.py", line 1536, in query
    check_status(response.status)
  File "/Users/zilliz/opt/anaconda3/envs/text_match/lib/python3.10/site-packages/pymilvus/client/utils.py", line 63, in check_status
    raise MilvusException(status.code, status.reason, status.error_code)
pymilvus.exceptions.MilvusException: <MilvusException: (code=1100, message=failed to create query plan: cannot parse expression: TextMatch(age, '10'), error: text match operation on non-string is unsupported: invalid parameter)>
 (api_request.py:45)
[2024-10-08 14:50:58 - ERROR - ci_test]: (api_response) : <MilvusException: (code=1100, message=failed to create query plan: cannot parse expression: TextMatch(age, '10'), error: text match operation on non-string is unsupported: invalid parameter)> (api_request.py:46)
```

**Since the text match operation is not supported on non-string data, why can't it directly report an error when creating a collection?**

**Comment by zhuwenxing** [comment]
/reopen

@liliu-z 
still reproduced in master-20241111-a88fff13-amd64 + pymilvus 2.5.0rc118

**Comment by liliu-z** [comment]
Some context:
1. We just ignore the `enable_match` for columns other than `VARCHAR`, instead of doing any mis-operation. Also, the error reported has nothing to do with whether adding `enable_match` to that column, but due to the datatype
2. In general there are two types of solution:
    1. Reject all useless param, in this case, all datatype specific params (I believe there are some for ARRAY, JSON, etc.) need to be handled
    2. Ignore all useless param, just like what we are.
3. Use index params as example, we are using solution2.

In conclusion, I thought it is a by design scenario instead of a bug.

**Comment by liliu-z** [comment]
@yanliang567 @zhuwenxing

**Comment by zhuwenxing** [comment]
Yes, from the final result, we reported an error for the misuse of using text match on non-varchar fields, which is logical.

The discussion here is actually about the timing of error reporting, whether it's fail fast or report errors during actual execution.

My idea is that enable_match is a parameter that only takes effect on varchar data types, so it would be better to convey this limitation to users as early as possible, which is also the approach in solution 1.

Of course, this is a tradeoff, and solution 1 requires adding many case-by-case checks.
